packages/flutter_tools/test/commands.shard/hermetic/doctor_test.dart

```diff
@@ -736,6 +753,7 @@ void main() {
       contains(isA<WebWorkflow>()));
   }, overrides: <Type, Generator>{
     FeatureFlags: () => TestFeatureFlags(isWebEnabled: true),
+    FileSystem: () => MemoryFileSystem.test(),
     ProcessManager: () => MockProcessManager(),
```

Comment:
Does it make sense for the context override to check for the inverse (ProcessManager and no FileSystem)?
https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/src/context.dart#L64

Reply:
I can't think of any reason not to.

Reply:
I guess the idea is if the process manager is mocked, then we shouldn't be hitting the file system at all?

Reply:
@jonahwilliams or @zanderso can you think of a reason?

Reply:
For context, I didn't change this because I knew for a fact it was hitting the file system, I just added it for consistency and since some of the other tests were hitting surprising globals.

Reply:
The real `ProcessManager` will certainly touch the real file system, subverting the intent of the `FileSystem` context override. The other way around doesn't have the same problem, but it could make sense to require always overriding both together just for consistency.